magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/update_MAGIC_database.py

```diff
@@ -0,0 +1,351 @@
+"""
+The script updates the common MAGIC database from a given time range.
+At the moment, to avoid accidentally destroying the previous database,
+we save the updated database as a new file. If the path to the database is not found,
+the script creates a new one. The start of the time interval
+is the date of the beginning of the common MAGIC+LST1 observations.
+The end of the time interval is the current date.
+
+The MAGIC files that can be used for analysis are located in the IT cluster
+in the following directory:
+/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}
+
+In this path, 'tel_id' refers to the telescope ID, which must be either 1 or 2.
+'YYYY', 'MM', and 'DD' specify the date.
+"""
+
+import os
+from datetime import datetime, timedelta
+
+import pandas as pd
+
+
+def fix_lists_and_convert(cell):
+    """
+    An additional function necessary to organize lists in the function table_magic_runs.
+    The function remove brackets to avoid double lists and split on "][".
+
+    Parameters
+    -----------
+    cell : str
+        List of MAGIC runs from the date and the source.
+
+    Returns
+    -------
+    list
+        A list of unique integers representing the MAGIC runs.
+    """
+
+    parts = cell.replace("][", ",").strip("[]").split(",")
+    return list(dict.fromkeys(int(item) for item in parts))
+
+
+def table_magic_runs(df, date_min, date_max):
+
+    """
+    Generate a table with data filtered by the specified date range.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        DataFrame with general information about MAGIC+LST1 observations.
+    date_min : str
+        Start of the time interval (in LST convention).
+    date_max : str
+        End of the time interval (in LST convention).
+
+    Returns
+    -------
+    pandas.DataFrame
+        A DataFrame filtered by the specified date range.
+    """
+
+    df_selected_data = df.iloc[:, [2, 1, 25]]
+    df_selected_data.columns = ["DATE", "source", "MAGIC_runs"]
+    grouped_data = df_selected_data.groupby(["DATE", "source"])
+    result_table = []
+
+    for (date, source), group in grouped_data:
+        if date >= date_min and date <= date_max: 
+            runs_combined = group["MAGIC_runs"].sum()
```

Comment:
did you check if this does not add the same MAGIC run multiple times if it is mentioned in different LST runs ?